zebra-state/src/service/finalized_state.rs

```diff
@@ -390,6 +396,19 @@ impl FinalizedState {
                 batch.zs_insert(history_tree_cf, height, history_tree);
             }
 
+            // Consensus rule: The block height of the genesis block is 0
+            // https://zips.z.cash/protocol/protocol.pdf#blockchain
+            if height == block::Height(0) {
+                batch.zs_insert(tip_chain_value_pool, height, ValueBalance::zero());
+            } else {
+                let current_pool = self.current_value_pool().constrain::<NegativeAllowed>()?;
+                batch.zs_insert(
+                    tip_chain_value_pool,
+                    height,
+                    (current_pool + block_value_balance)?.constrain::<NonNegative>()?,
+                );
```

Comment:
This looks a lot like `ValueBalance::update_with_chain_value_pool_change`.
Can you use that instead?

Reply:
https://github.com/ZcashFoundation/zebra/pull/2599/commits/abd23b0835f01817089a87d644ddff9da1adcb30